### PR TITLE
[css-fonts-4] Animating palettes

### PR DESCRIPTION
**Pull request by drott**
Draft wording to address #8922, created as a PR against CSS Fonts 4 to 
make it easier to see differences. I would expect this to be an addition to CSS Fonts 5.

Define `palette-mix()` function, interpolation and percentage normalization behavior
by help of referencing CSS Color Module 5.

### CODE REVIEW

**Comment by drott** [comment]
@tabatkins @svgeesus Could I ask you to review this draft? Our hope is that this is something we can present as a as basis for discussion and proposal for resolution at TPAC. Thanks in advance.

**Comment by svgeesus** [comment]
On it.

File: css-fonts-4/Overview.bs [hunk 1]
```diff
@@ -7058,6 +7064,25 @@ Animation type: discrete
...
+        identified by <<palette-identifier>>, by the palette keywords
+        ''font-palette/light'' or ''font-palette/dark'',
+        or by another <<palette-mix()>> value.
+
+        The glyphs of the font must be rendered with a palette that is created by
+        interpolating for each palette color index between the matching color indices
+        of the first and second palette specified as arguments,
+        recursively resolving <<palette-mix()>> functions if needed.
+
+        For each palette color, the interpolation follows
```
Comment (svgeesus) [inline css-fonts-4/Overview.bs:7079]: Does this just mean that it uses the [&lt;color-interpolation-method>](https://drafts.csswg.org/css-color-4/#color-interpolation-method) token and in general follows CSS Color 4 [Color Interpolation](https://drafts.csswg.org/css-color-4/#interpolation) or is there a tighter dependency on `color-mix()`, for example normalization of percentage values?

File: css-fonts-4/Overview.bs [hunk 2]
```diff
@@ -7058,6 +7064,25 @@ Animation type: discrete
...
+        the rules of the <<color-mix()>> function.
+
+        When not specified otherwise the <<color-interpolation-method>> is
+        <code>in ''oklab''</code> as recommended by [[css-color-5#interpolation-space]].
+
+        Percentages are normalized according to [[css-color-5#color-mix-percent-norm]].
```
Comment (svgeesus) [inline css-fonts-4/Overview.bs:7085]: Ah okay.

File: css-fonts-4/Overview.bs [hunk 3]
```diff
@@ -7068,7 +7093,39 @@ Note: The names 'light' and 'dark' describe the <i>background</i> the color pale
...
+    }
+}
+p {
+     font-family: Nabla;
+     animation: animate-palette 1.4s infinite alternate linear;
+}
+    </pre>
+
+	<figure>
+		<img src="images/nabla-animated.webp" alt="The word Animate animated between a pink and a yellow palette."/>
```
Comment (svgeesus) [inline css-fonts-4/Overview.bs:7126]: Nice example

**Review by svgeesus — APPROVED** [review]
This all looks great, very clear, lets discuss and get a resolution at TPAC.

**Comment by svgeesus** [comment]
I appreciate the suggestion to put this in Fonts 5, but wonder if it would be clearer in Fonts 4 to keep all the changes in one place and avoid duplication. I could go either way.

File: css-fonts-4/Overview.bs [hunk 4]
```diff
@@ -7058,6 +7064,25 @@ Animation type: discrete
 	    be a modification of an existing palette already present in the font,
 	    or be an entirely new palette.
 
+	<dt><dfn type><<palette-mix()>></dfn>
+	<dd>
+        This value defines a position in the interpolation between two palette values,
+        identified by <<palette-identifier>>, by the palette keywords
+        ''font-palette/normal'', ''font-palette/light'', ''font-palette/dark'',
```
Comment (drott) [inline css-fonts-4/Overview.bs:7071]: FYI @tursunova - `normal` now added, thanks for pointing that out.

File: css-fonts-4/Overview.bs (hunk 1 above)
Comment (drott) [reply css-fonts-4/Overview.bs:7079]: Yes, as you noticed, percentage normalization specified  below.

File: css-fonts-4/Overview.bs [hunk 5]
```diff
@@ -6955,13 +6955,14 @@ as well as the colored versions.
...
-Value: normal | light | dark | <<palette-identifier>>
+Value: normal | light | dark | <<palette-identifier>> | <<palette-mix()>>
 Initial: normal
 Applies to: all elements and text
 Inherited: yes
 Percentages: N/a
-Computed value: specified keyword or identifier
-Animation type: discrete
+Computed value: specified keyword, identifier or <<palette-mix()>> function.
+    <<palette-mix>> must be simplified to a single keyword or identifier if resulting palette is equivalent.
```
Comment (drott) [inline css-fonts-4/Overview.bs:6964]: @lilles Thanks for the feedback, now updated.

**Comment by drott** [comment]
> I appreciate the suggestion to put this in Fonts 5, but wonder if it would be clearer in Fonts 4 to keep all the changes in one place and avoid duplication. I could go either way.

Agree, I would also prefer an edit to Fonts 4 if possible.

**Pull request merged**